### ISSUE TITLE
Enable GKE workflow to load input files of various sizes without crashing

### ISSUE THREAD

**Issue by julietcohen** [PermafrostDiscoveryGateway/viz-workflow#44]
The GKE workflow successfully ingests files and writes files when it processes the two smallest IWP files our team uses for testing, which are only 18-20 MB each. These 2 files do not have footprints and are solely used for quickly testing the basic functionality of the workflow.

However, when the GKE workflow is executed with _more_ input files and/or _larger_ input files, parsl errors. This can be replicated by using the archived sample of 3 adjacent IWP files with footprints that are derived from the pan-Arctic IWP dataset. The most recent built image with the files configured for this is [version 0.2.9-test-2](https://github.com/PermafrostDiscoveryGateway/viz-workflow/pkgs/container/viz-workflow/251006662?tag=0.2.9-test-2). These 3 files and their footprints are frequently used to test the workflow for deduplication or scaling up in general. The polygons in these files cover a very small area on Wrangle Island and are the same files used in the `viz-info/helpful-code` repo to help onboard team members to the viz workflow. Knowing where the output tilesets should appear on a cesium map makes it easy to check the web tile output for this data sample. In total, the size of this input dataset is still small: 1.4 GB. Both samples of IWP data are available in the shared [Google Drive](https://drive.google.com/drive/u/0/folders/0AEcXOj3BkIRlUk9PVA) that contains dataset samples for those who do not have access to Datateam. Of course, since I already tested the GKE workflow with these input files, they have already been uploaded to the storage bucket [here](https://console.cloud.google.com/storage/browser/pdg-storage-default/viz_workflow/input_3_iwp_fp?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&project=pdg-project-406720).

When the GKE workflow is executed with this larger sample of 3 IWP files and their footprints, the workflow starts up fine and fires up 4 connected workers. No files are written to the output dir. After a few minutes, the terminal outputs the following:
<img width="1478" alt="Pasted Graphic 4" src="https://github.com/user-attachments/assets/d4716b0a-53ff-47e9-9cc6-7c35a0627049">

**Comment by julietcohen** [comment]
Possible solutions:

- check out the pod logs to see if reading in a large file is indeed the issue
- increase the time allocated to pods to launch or workers to initialize with additional parameter specifications in the parsl config:
  - `pod_launch_timeout = {LARGE NUMBER}` (units for this number? seconds?)
  - `worker_init_timeout = {LARGE NUMBER}`
  -  `heartbeat_threshold = {LARGE NUMBER`
- increase resource allocation with other config parameters:
  - `init_mem='{NUMBER}Gi'`
  - `max_mem='{NUMBER}Gi',`
- utilize "persistent workers" to improve performance with long tasks or large files using the `persistent_worker_threshold = 0` parameter 
  - I haven't worked with these before, but apparently they're able to stay alive after completing a task (in order to accept a new task and eliminate the overhead of spinning up a new worker)
  - can retain the large dataset loaded in to use it for multiple tasks

**Comment by julietcohen** [comment]
I ran the workflow with the 3 larger IWP files and footprints sampled from the actual IWP dataset, with parameters in the config to give the pods more time to load in the files:

```python
# increased seconds allocated towards a task before it is 
# considered failed,for loading in large files:
heartbeat_threshold = 3600,
# default seconds for how often a heartbeat is sent:
heartbeat_period = 30, 
```

Published as v0.3.0 in the repo.

The pods ran much longer, but still errored:

<img width="1067" alt="image" src="https://github.com/user-attachments/assets/c779c2bc-d9c2-4d8f-86eb-3e65918b6998">

**Comment by julietcohen** [comment]
Including parameters for both memory and time for tasks did the trick. This is reflected in v0.3.1. Output files for the 3 IWP and footprints were written to the bucket.

Here's the whole parsl config:


<details>
  <summary>config_parsl_cluster</summary>

```python
def config_parsl_cluster(
        max_blocks = 4,
        min_blocks = 1,
        init_blocks = 1,
        max_workers = 4,
        cores_per_worker = 1,
        # TODO: automate this following string to pull most recent release on github?
        image='ghcr.io/permafrostdiscoverygateway/viz-workflow:0.3.1',
        namespace='viz-workflow'):

    htex_kube = Config(
        executors = [
            HighThroughputExecutor(
                label = 'kube-htex',
                cores_per_worker = cores_per_worker,
                max_workers = max_workers,
                worker_logdir_root = '/',
                # Address for the pod worker to connect back
                address = address_by_route(),
                worker_debug = True,
                # increased seconds allocated towards a task before it is 
                # considered failed,for loading in large files:
                heartbeat_threshold = 3600,
                # default seconds for how often a heartbeat is sent:
                heartbeat_period = 30, 
                provider = KubernetesProvider(

                    # Namespace in K8S to use for the run
                    namespace = namespace,

                    # Docker image url to use for pods
                    image = image,

                    # Should follow the Kubernetes naming rules
                    pod_name = 'viz-workflow-worker',

                    init_mem='4Gi',
                    max_mem='8Gi',

                    nodes_per_block = 1,
                    init_blocks = init_blocks,
                    min_blocks = min_blocks,
                    # Maximum number of pods to scale up
                    max_blocks = max_blocks,
                    # persistent_volumes (list[(str, str)]) – List of tuples
                    # describing persistent volumes to be mounted in the pod.
                    # The tuples consist of (PVC Name, Mount Directory).
                    persistent_volumes = [('viz-workflow-pvc', f'/data')],

                    # This annotation is required to mount a GCS PVC to the pod and
                    # the service account (with sufficient permissions) is required
                    # to access a GCS PVC.
                    annotations={"gke-gcsfuse/volumes": "true"},
                    service_account_name="viz-workflow-sa",
                ),
            ),
        ]
    )
    return(htex_kube)
```
</details>
